pipeline-runner/R/handle_data.R

```diff
@@ -1,16 +1,56 @@
-reload_scdata_from_s3 <- function(pipeline_config, experiment_id) {
+upload_cells_id <- function(pipeline_config, object_key, cells_id) {
+  object_data <- tempfile()
+  saveRDS(cells_id, file = object_data)
+  put_object_in_s3(pipeline_config, pipeline_config$cells_id_bucket, object_data, object_key)
+  return(object_key)
+}
+
+
+reload_scdata_from_s3 <- function(pipeline_config, experiment_id, task_name, tasks) {
+  # If the task is after data integration, we need to get scdata from processed_matrix
+  task_names <- names(tasks)
+  integration_index <- which(task_names == "dataIntegration")
+  if (match(task_name, task_names) > integration_index) {
+    bucket <- pipeline_config$processed_bucket
+  } else {
+    bucket <- pipeline_config$source_bucket
+  }
   s3 <- paws::s3(config = pipeline_config$aws_config)
-  message(pipeline_config$source_bucket)
+  message(bucket)
   message(paste(experiment_id, "r.rds", sep = "/"))
 
   c(body, ...rest) %<-% s3$get_object(
-    Bucket = pipeline_config$source_bucket,
+    Bucket = bucket,
     Key = paste(experiment_id, "r.rds", sep = "/")
   )
   obj <- readRDS(rawConnection(body))
   return(obj)
 }
 
+load_cells_id_from_s3 <- function(pipeline_config, task_name, experiment_id, samples) {
+  s3 <- paws::s3(config = pipeline_config$aws_config)
+  object_list <- s3$list_objects("biomage-filtered-cells-development", Prefix = paste0(experiment_id, "/", task_name, "/"))
```

Comment:
add a check either here or outside functions that we retrieved cell ids for all samples in scdata